### PR TITLE
Update Descartes and OMPL to leverage new Kinematics objects

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR also address clang-tidy warnings taken from #312

### CODE REVIEW

**Comment by mpowelson** [comment]
Do you want to squash some of these commits before we merge?

File: tesseract/tesseract_planning/tesseract_motion_planners/CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +77,9 @@ target_include_directories(${PROJECT_NAME}_ompl PUBLIC
     "$<INSTALL_INTERFACE:include>")
 target_include_directories(${PROJECT_NAME}_ompl SYSTEM PUBLIC
     ${OMPL_INCLUDE_DIRS})
+if(OMPL_VERSION VERSION_LESS "1.4.0")
+    target_compile_definitions(${PROJECT_NAME}_ompl PUBLIC OMPL_LESS_1_4_0=ON)
+endif()
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/CMakeLists.txt:82]: Could you enable the TESSERACT_ENABLE_TESTING flag again? Somewhere it got commented out so tests are always run

Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/CMakeLists.txt:82]: fixed.

**Comment by Levi-Armstrong** [comment]
If you are good with it I will squash the code coverage and clang format commits into a single commit?

**Comment by mpowelson** [comment]
Yeah that's fine with me.

**Comment by Levi-Armstrong** [comment]
replaced by #312

**Pull request closed without merging**